### ISSUE TITLE
CA1305 false positive on Nullable<Int32>

### ISSUE THREAD

**Issue by avivanoff** [dotnet/roslyn-analyzers#7054]
### Analyzer

**Diagnostic ID**: [CA1305](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2013): \`Specify IFormatProvider\`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 8 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100](https://dotnet.microsoft.com/download/dotnet/8.0)

### Steps To Reproduce

1. Create SDK-style project that targets .NET Framework 4.8 and enable NET Analyzers.
2. Make sure CA1305 is enabled.
3. Add code that call ToString on nullable Int32.
4. Build the project.

### Expected behavior
CA1305 should not fire.

### Actual behavior
CA1305 is triggered.

**Comment by arphox** [comment]
I also found an "unsolvable" report of CA1305.
I have a `double?` local for which I call `.ToString()` and CA1305 reports that I need to add an IFormatProvider to the ToString method, but the problem is that method doesn't have that kind of overload :D

**Comment by CollinAlpert** [comment]
That is probably since you are calling `ToString()` directly and not using null-propagation. This works:
```csharp
public void Run(int? i)
{
    var s = i?.ToString(CultureInfo.InvariantCulture);
}
```

**Comment by avivanoff** [comment]
This calls ToString on the wrapped instance, not [Nullabe<T>.ToString](https://www.bestbuy.com/site/samsung-t7-1tb-external-usb-3-2-gen-2-portable-ssd-with-hardware-encryption-titan-gray/6408299.p?skuId=6408299) that does not have an overload that accepts IFormatProvider.

**Comment by sharwell** [comment]
This is almost certainly by design, since it would otherwise be far too easy to accidentally call the wrong method and omit the format provider.

**Comment by sharwell** [comment]
Duplicate of #6586

**Comment by sharwell** [comment]
Not exactly a duplicate, but more "by design as shown by #6586"

**Comment by avivanoff** [comment]
> This is almost certainly by design, since it would otherwise be far too easy to accidentally call the wrong method and omit the format provider.

@sharwell, what if I do want to call Nullable<T>.ToString? Could you point me design document that describes the behavior?

**Comment by sharwell** [comment]
@avivanoff it's the same as if you want to call `int.ToString()`. You'd suppress the warning at the location where you want to make the call.

**Comment by avivanoff** [comment]
@sharwell, how about a link to design documents?

**Comment by sharwell** [comment]
The reference link for this behavior #6586
